### PR TITLE
fix traffic accounting in internal statistics / metrics

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Backport of https://github.com/arangodb/arangodb/pull/17651

Fix traffic accounting for HTTP/VST in internal statistics / metrics.
Outgoing VST traffic was not accounted for previously.
Additionally, the traffic was not accounted for the request that initiates a protocol change from HTTP/1.1 to HTTP/2.
Finally, the metrics histograms that summed up traffic were not correctly reporting values since 3.10.

For consistency, this PR also adds a new metrics `arangodb_http1_connections_total`, which counts the number of HTTP/1.1 connections. There were previously only metrics for HTTP/2 and VST connections.

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [x] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [x] **integration tests**
  - [ ] **resilience tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [x] Backports
  - [x] Backport for 3.10: https://github.com/arangodb/arangodb/pull/17652
  - [x] Backport for 3.9: https://github.com/arangodb/arangodb/pull/17653
  - [x] Backport for 3.8: this PR

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/GeneralServer/CommTask.cpp [hunk 1]
```diff
@@ -420,6 +420,11 @@ void CommTask::executeRequest(std::unique_ptr<GeneralRequest> request,
 // --SECTION-- statistics handling                             protected methods
 // -----------------------------------------------------------------------------
 
+void CommTask::setStatistics(uint64_t id, RequestStatistics::Item&& stat) {
+  std::lock_guard<std::mutex> guard(_statisticsMutex);
```
Comment (MBkkt) [inline arangod/GeneralServer/CommTask.cpp:424]:
```suggestion
  std::lock_guard guard{_statisticsMutex};
```

File: arangod/GeneralServer/CommTask.h [hunk 2]
```diff
@@ -150,18 +158,17 @@ class CommTask : public std::enable_shared_from_this<CommTask> {
  private:
   bool handleRequestSync(std::shared_ptr<RestHandler>);
   bool handleRequestAsync(std::shared_ptr<RestHandler>, uint64_t* jobId = nullptr);
-  
+
+  mutable std::mutex _statisticsMutex;
+  std::unordered_map<uint64_t, RequestStatistics::Item> _statisticsMap;
```
Comment (MBkkt) [inline arangod/GeneralServer/CommTask.h:163]: containers::FlatHashMap?

File: arangod/GeneralServer/VstCommTask.cpp [hunk 3]
```diff
@@ -455,15 +455,14 @@ void VstCommTask<T>::doWrite() {
...
+                        rsp(std::move(item))](asio_ns::error_code const& ec, size_t nwrite) {
                          DTraceVstCommTaskAfterAsyncWrite((size_t)self.get());
 
                          auto& me = static_cast<VstCommTask<T>&>(*self);
                          me._writing = false;
 
                          rsp->stat.SET_WRITE_END();
-                         rsp->stat.ADD_SENT_BYTES(rsp->buffers[0].size() +
-                                                  rsp->buffers[1].size());
+                         rsp->stat.ADD_SENT_BYTES(nwrite);
```
Comment (MBkkt) [inline arangod/GeneralServer/VstCommTask.cpp:465]: assert(nwrite == ?)?

**Review by MBkkt — APPROVED** [review]

**Review by neunhoef — APPROVED** [review]
LGTM

**Comment by jsteemann** [comment]
Jenkins run (not automatically picked up): https://jenkins.arangodb.biz/view/PR/job/arangodb-matrix-pr/25935/

**Pull request merged**